### PR TITLE
Improve error messages

### PR DESCRIPTION
**Pull request by rcannood**
## Changes

* Added unit test for the InstanceAPI class (PR #30).

* Add alternative error message when no message is returned from the API (PR #30).

### CODE REVIEW

File: R/InstanceAPI.R [hunk 1]
```diff
@@ -19,21 +19,16 @@ InstanceAPI <- R6::R6Class( # nolint object_name_linter
...
+        "/instances/",
+        private$.instance_settings$id,
+        "/schema"
       )
 
-      content <- httr::content(request)
-      if (httr::http_error(request)) {
-        cli_abort(content$detail)
-      }
+      request <- httr::GET(url)
```
Comment (lazappi) [inline R/InstanceAPI.R:29]: Should this be "response" rather than "request"?

File: R/InstanceAPI.R [hunk 2]
```diff
@@ -96,15 +91,22 @@ InstanceAPI <- R6::R6Class( # nolint object_name_linter
...
+    .instance_settings = NULL,
+    process_request = function(request, request_type) {
       content <- httr::content(request)
       if (httr::http_error(request)) {
-        cli_abort(content$detail)
+        if (is.list(content) && "detail" %in% names(content)) {
+          cli_abort(content$detail)
+        } else {
+          cli_abort(paste0("Failed to ", request_type, " from instance. Response output: ", content))
+        }
```
Comment (lazappi) [inline R/InstanceAPI.R:106]: I'm not sure how this will work with the API package but we can sort that out later

File: R/InstanceSettings.R [hunk 3]
```diff
@@ -37,13 +37,13 @@ InstanceSettings <- R6::R6Class( # nolint object_name_linter
         "db_user_password", # api
         "lamindb_version" # api
       )
-      missing_column <- setdiff(expected_keys, names(settings))
-      if (length(missing_column) > 0) {
-        cli_abort("Missing column: ", missing_column)
+      missing_key <- setdiff(expected_keys, names(settings))
+      if (length(missing_key) > 0) {
+        cli_abort(paste0("Missing key: ", paste(missing_key, collapse = ", ")))
```
Comment (lazappi) [inline R/InstanceSettings.R:42]: Missing key**s**? I think **{cli}** can automatically pluralise stuff if you want to get fancy but not sure it's worth the effort.

File: tests/testthat/test-instance_api.R [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+      schema_str = "foo,bar",
+      schema_id = "...",
+      git_repo = "...",
+      keep_artifacts_local = TRUE,
+      api_url = "https://foo.lamin.ai"
+    )
+  )
+}
+
+test_that("test get_schema", {
```
Comment (lazappi) [inline tests/testthat/test-instance_api.R:18]: Definitely not important but I think you are supposed to read this as a sentence starting with "Test that..." so the extra "test" is a bit redundant.

**Review by lazappi — APPROVED** [review]

File: R/InstanceAPI.R (hunk 2 above)
Comment (rcannood) [reply R/InstanceAPI.R]: agreed

File: R/InstanceAPI.R (hunk 1 above)
Comment (rcannood) [reply R/InstanceAPI.R]: good point

File: R/InstanceSettings.R (hunk 3 above)
Comment (rcannood) [reply R/InstanceSettings.R]: Oh, this is cool, I didn't know `{cli}` could do this!

File: tests/testthat/test-instance_api.R (hunk 4 above)
Comment (rcannood) [reply tests/testthat/test-instance_api.R]: fixed!

**Review by lazappi — APPROVED** [review]

File: R/InstanceSettings.R [hunk 5]
```diff
@@ -37,13 +37,13 @@ InstanceSettings <- R6::R6Class( # nolint object_name_linter
         "db_user_password", # api
         "lamindb_version" # api
       )
-      missing_column <- setdiff(expected_keys, names(settings))
-      if (length(missing_column) > 0) {
-        cli_abort("Missing column: ", missing_column)
+      missing_keys <- setdiff(expected_keys, names(settings))
+      if (length(missing_key) > 0) {
```
Comment (lazappi) [inline R/InstanceSettings.R]: Missing a `s` here

```suggestion
if (length(missing_keys) > 0) {
```

Comment (rcannood) [reply R/InstanceSettings.R]: Would accept this suggestion but already fixed it myself based on the feedback you gave

**Pull request merged**